pytorch_lightning/core/datamodule.py

```diff
@@ -257,3 +257,19 @@ def test_dataloader():
         if test_dataset is not None:
             datamodule.test_dataloader = test_dataloader
         return datamodule
+
+    def state_dict(self) -> Dict[str, Any]:
+        """Called when saving a checkpoint, implement to generate and save datamodule state.
+
+        Returns:
+            A dictionary containing datamodule state.
+        """
+        return {}
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
```

Comment:
```suggestion
    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
```
quick ques: does `Dist` supports `OrderedDict`? since, for PT Module, it is OrderedDict. Or should we use `Mapping`

Reply:
yep checked mypy has no issues with this:

```
def testingmethod() -> Dict[Any, Any]:
    return OrderedDict()
```